witnesslint.py

```diff
@@ -0,0 +1,1128 @@
+"""
+This module contains a linter that can check witnesses for consistency
+with the witness format [1].
+
+[1]: github.com/sosy-lab/sv-witnesses/blob/master/README.md
+"""
+
+import argparse
+import collections
+import gzip
+import hashlib
+import logging
+import re
+import sys
+import time
+from lxml import etree as ET
+
+COMMON_KEYS = {
+    "witness-type": "graph",
+    "sourcecodelang": "graph",
+    "producer": "graph",
+    "specification": "graph",
+    "programfile": "graph",
+    "programhash": "graph",
+    "architecture": "graph",
+    "creationtime": "graph",
+    "entry": "node",
+    "sink": "node",
+    "violation": "node",
+    "invariant": "node",
+    "invariant.scope": "node",
+    "assumption": "edge",
+    "assumption.scope": "edge",
+    "assumption.resultfunction": "edge",
+    "control": "edge",
+    "startline": "edge",
+    "endline": "edge",
+    "startoffset": "edge",
+    "endoffset": "edge",
+    "enterLoopHead": "edge",
+    "enterFunction": "edge",
+    "returnFromFunction": "edge",
+    "threadId": "edge",
+    "createThread": "edge",
+}
```

Comment:
I have noticed that the keys of this dict are also used later on in the code for comparison.

Please use constants for these strings.

Reply:
Done.